libyul/ObjectParser.cpp

```diff
@@ -104,10 +127,92 @@ shared_ptr<Block> ObjectParser::parseCode()
 	return parseBlock();
 }
 
+optional<ObjectParser::ReverseSourceNameMap> ObjectParser::tryGetSourceLocationMapping() const
+{
+	return tryGetSourceLocationMapping(m_scanner->currentCommentLiteral());
+}
+
+optional<ObjectParser::ReverseSourceNameMap> ObjectParser::tryGetSourceLocationMapping(std::string const& _text)
+{
+	// @use-src 0:"abc.sol" , "1:foo.sol" ,2:"bar.sol"
+	//
+	// UseSrcList := UseSrc (',' UseSrc)*
+	// UseSrc     := [0-9]+ ':' FileName
+	// FileName   := "(([^\"]|\.)*)"
+
+	// Matches some "@use-src TEXT".
+	static std::regex const lineRE = std::regex(
+		R"~~~((^|\s+)@use-src\s+(.*)$)~~~",
+		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	);
+	string_view text(_text);
+	std::cmatch cm;
+	if (!std::regex_search(text.data(), text.data() + text.size(), cm, lineRE))
+		return nullopt;
+	solAssert(cm.size() == 3, "");
+
+	// Let @c text point to the parameter value (last match).
+	text = string_view(cm[2].first, static_cast<size_t>(std::distance(cm[2].first, cm[2].second)));
```

Comment:
Why don't you just take `c[2]` directly?

Reply:
`cm` is a C-string `std::sub_match`, using `cm[2].str()` would construct a new `std::string`. `cm[2].first` and `cm[2].end` are basically like `begin` and `end` toe the sub-match range. This way and with `string_view.remove_prefix(...)` I could work on the same text sequence without any temporary copies.

Reply:
I have changed that to use `string` instead of `string_view` now. I am not sure what version is better. the `string` version does less offseting, the `string_view` version has additional allocations.